### DISCUSSION TITLE
WPF Cef make browser window to not reload every time view changed

### DISCUSSION THREAD

**Discussion by TimurVa** [cefsharp/CefSharp#3862] [Q&A] [answered]
Hi there. I have an issue. I'm creating cefbrowser in my wpf application using MVVM line this:
                <cef:ChromiumWebBrowser Address="{Binding Address}"
                                        Title="{Binding BrowserTitle}"
                                        RenderOptions.BitmapScalingMode="HighQuality"
                                        WebBrowser="{Binding WebBrowser, Mode=OneWayToSource}">
I have something like "TabControl" with items and in one tab there is this browser. So every time i'm changing tab or page inside my wpf application and returning back to tab where Browser was - Browser is refreshing. Is there any way I can stick with the same Browser on the same page and nothing could be reloaded? I have consistent data and using signalR, so this is crucial for me. Any help please?

**Comment by amaitland** [comment, marked as answer]
You can use a custom tabcontrol see https://github.com/cefsharp/CefSharp/blob/master/CefSharp.Wpf.Example/Controls/NonReloadingTabControl.cs#L21 for the basic implementation used by the WPF example. 

There are plenty of articles that cover the topic.
- https://www.codeproject.com/Articles/460989/WPF-TabControl-Turning-Off-Tab-Virtualization
- https://stackoverflow.com/questions/3601125/wpf-tabcontrol-preventing-unload-on-tab-change

**Comment by TimurVa** [comment]
I found out that if I use ContentControl with Content binding to my WebBrowser - everything works as expected. The problem with custom tab control is - I'm using AvalonDock so there is I think no way I can change source code. Also I'm using Frame which is changin Pages inside application which is Also calls unloaded for whole page. I thought there is a property or something else than can help me. Any way thanks for mension of custom control maybe I'll find something usefull from that idea.
